crates/binjs_es6/build.rs

```diff
@@ -9,23 +9,33 @@ use std::env;
 use std::fs::*;
 use std::io::*;
 
-const PATH_GRAMMAR_ES6: &'static str = "../../spec/es6.webidl";
+/// The webidl source files.
+///
+/// Order is important: first the main file, then any extension.
+const PATH_SOURCES: [&'static str; 2] =
+    ["../../spec/es6.webidl", "../../spec/es6-extension.webidl"];
 
 fn main() {
-    println!("cargo:rerun-if-changed={}", PATH_GRAMMAR_ES6);
+    for source in &PATH_SOURCES {
+        println!("cargo:rerun-if-changed={}", source);
+    }
 
     // Load webidl.
 
-    let mut file = File::open(PATH_GRAMMAR_ES6).expect("Could not open source");
-    let mut source = String::new();
-    file.read_to_string(&mut source)
-        .expect("Could not read source");
+    let sources: Vec<_> = PATH_SOURCES
```

Comment:
Now that iterators are supported, I don't think you need intermediate vectors anymore?

Reply:
Ah... you might need them for the reference to point somewhere. Could get away with more generics, but probably not worth it at this point.